host_core/native/hostcore_wasmcloud_native/src/oci.rs

```diff
@@ -50,32 +53,57 @@ pub(crate) async fn fetch_oci_path(
             "Fetching images tagged 'latest' is currently prohibited in this host. This option can be overridden with WASMCLOUD_OCI_ALLOW_LATEST".into());
     }
     let cf = cached_file(img).await?;
-    if tokio::fs::metadata(&cf).await.is_err() {
-        let auth = determine_auth(img, creds_override);
-        let img = oci_distribution::Reference::from_str(img)?;
 
-        let protocol =
-            oci_distribution::client::ClientProtocol::HttpsExcept(allowed_insecure.to_vec());
-        let config = oci_distribution::client::ClientConfig {
-            protocol,
-            ..Default::default()
-        };
-        let mut c = oci_distribution::Client::new(config);
-        let imgdata = pull(&mut c, &img, &auth).await;
+    let auth = determine_auth(img, creds_override);
+    let img = oci_distribution::Reference::from_str(img)?;
+
+    let protocol =
+        oci_distribution::client::ClientProtocol::HttpsExcept(allowed_insecure.to_vec());
+    let config = oci_distribution::client::ClientConfig {
+        protocol,
+        ..Default::default()
+    };
+    let mut c = oci_distribution::Client::new(config);
 
+    if tokio::fs::metadata(&cf).await.is_err() {     
+        let imgdata = pull(&mut c, &img, &auth).await;
         match imgdata {
             Ok(imgdata) => {
-                let mut f = tokio::fs::File::create(&cf).await?;
-                let content = imgdata
-                    .layers
-                    .into_iter()
-                    .flat_map(|l| l.data)
-                    .collect::<Vec<_>>();
-                f.write_all(&content).await?;
-                f.flush().await?;
+                cache_oci_image(&imgdata, &cf).await;
             }
             Err(e) => return Err(format!("Failed to fetch OCI bytes: {}", e).into()),
         }
+    }else{
```

Comment:
I'm not sure if we have format checking set up in the NIF, but you'll want to run `cargo fmt -- --check && cargo clippy -- -D warnings` from the `hostcore_wasmcloud_native` directory

Reply:
Ahh missed that one.